### PR TITLE
Use trigger_deprecation instead of @trigger_error

### PR DESCRIPTION
**Pull request by l-vo**

### CODE REVIEW

**Comment by wouterj** [comment]
Thanks @l-vo!

Fyi, I've slightly modified the changes in this PR in a2846b9, to make some things a bit more dense. I've also removed the note about assert, as this has been removed in https://github.com/symfony/symfony/pull/35648

File: contributing/code/conventions.rst [hunk 1]
```diff
@@ -100,40 +100,38 @@ A feature is marked as deprecated by adding a ``@deprecated`` PHPDoc to
...
 
     /**
-     * @deprecated since Symfony 2.8, use A\B\Replacement instead.
+     * @deprecated since Symfony 5.1, use A\B\Replacement instead.
      */
 
 A PHP ``E_USER_DEPRECATED`` error must also be triggered to help people with the migration::
 
-    @trigger_error(sprintf('The "%s" class is deprecated since Symfony 2.8, use "%s" instead.', Deprecated::class, Replacement::class), E_USER_DEPRECATED);
+    trigger_deprecation('vendor-name/package-name', '5.1', 'The "%s" class is deprecated since Symfony 5.1, use "%s" instead.', Deprecated::class, Replacement::class);
```
Comment (nicolas-grekas) [inline contributing/code/conventions.rst:121]: `since Symfony 5.1` should be removed from the message

File: components/phpunit_bridge.rst [hunk 2]
```diff
@@ -163,13 +163,9 @@ Trigger Deprecation Notices
...
 
-    @trigger_error('Your deprecation message', E_USER_DEPRECATED);
+    trigger_deprecation('vendor-name/package-name', '5.1', 'Your deprecation message');
 
-Without the `@-silencing operator`_, users would need to opt-out from deprecation
-notices. Silencing by default swaps this behavior and allows users to opt-in
-when they are ready to cope with them (by adding a custom error handler like the
-one provided by this bridge). When not silenced, deprecation notices will appear
-in the **Unsilenced** section of the deprecation report.
+Where 5.1 is the version from which the deprecation starts. Note that the deprecation message can use the :phpfunction:`printf` format. In this case, you can pass placeholders as extra arguments after the deprecation message.
```
Comment (nicolas-grekas) [inline components/phpunit_bridge.rst:168]: missing quotes around 5.1 for clarity

File: contributing/code/conventions.rst [hunk 3]
```diff
@@ -100,40 +100,38 @@ A feature is marked as deprecated by adding a ``@deprecated`` PHPDoc to
...
 
-Without the `@-silencing operator`_, users would need to opt-out from deprecation
-notices. Silencing swaps this behavior and allows users to opt-in when they are
-ready to cope with them (by adding a custom error handler like the one used by
-the Web Debug Toolbar or by the PHPUnit bridge).
+The ``trigger_deprecation`` function internally use the php function :phpfunction:`assert`. This means you should use `zend.assertions` to disable deprecations in production.
```
Comment (nicolas-grekas) [inline contributing/code/conventions.rst:123]: uses (missing "s")
we don't use assert anymore actually

File: contributing/code/conventions.rst [hunk 4]
```diff
@@ -100,40 +100,38 @@ A feature is marked as deprecated by adding a ``@deprecated`` PHPDoc to
...
+When deprecating a whole class the ``trigger_deprecation()`` call should be placed
 between the namespace and the use declarations, like in this example from
 `ServiceRouterLoader`_::
 
     namespace Symfony\Component\Routing\Loader\DependencyInjection;
 
     use Symfony\Component\Routing\Loader\ContainerLoader;
 
-    @trigger_error(sprintf('The "%s" class is deprecated since Symfony 4.4, use "%s" instead.', ServiceRouterLoader::class, ContainerLoader::class), E_USER_DEPRECATED);
+    trigger_deprecation('symfony/routing', '4.4', 'The "%s" class is deprecated since Symfony 4.4, use "%s" instead.', ServiceRouterLoader::class, ContainerLoader::class);
```
Comment (nicolas-grekas) [inline contributing/code/conventions.rst:134]: "since etc" should be removed from the message

**Review by nicolas-grekas — COMMENTED** [review]
sorry for the late review

File: contributing/code/conventions.rst (hunk 1 above)
Comment (wouterj) [reply contributing/code/conventions.rst:121]: This was fixed while merging (a2846b9f04b371bc3f1f11f87075b2c686799861)

File: components/phpunit_bridge.rst (hunk 2 above)
Comment (wouterj) [reply components/phpunit_bridge.rst:168]: This section was also changed during merge (a2846b9f04b371bc3f1f11f87075b2c686799861)

**Comment by nicolas-grekas** [comment]
Perfect then, thanks!

**Pull request merged**